crates/swc_ecma_fast_parser/src/lexer/cursor.rs

```diff
@@ -0,0 +1,341 @@
+//! Byte-level cursor for fast input traversal
+//!
+//! This cursor operates directly on UTF-8 bytes for maximum performance.
+
+use swc_common::BytePos;
+use wide::u8x16;
+
+use crate::util::{likely, unlikely};
+
+/// High-performance cursor for traversing input bytes
+#[repr(C)] // Ensure predictable memory layout for better cache behavior
+pub struct Cursor<'a> {
+    /// Input source as bytes
+    input: &'a [u8],
+
+    /// Current position in bytes
+    pos: usize,
```

Comment:
Would `u32` be better here? It natively handles JavaScript inputs up to 4GB and avoids unnecessary casts, making the data flow more explicit. Thoughts?

Reply:
Yeap. I missed it. Thank you!
(I forgot it while doing `git reset --hard` due to regression)

Reply:
The point is that, `pos` and `len` can be fetched at once on 64 bit machines